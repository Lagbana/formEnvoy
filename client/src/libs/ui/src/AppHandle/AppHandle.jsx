/* Handle
=============================== */

import React from 'react';
import { View, StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';

export interface AppHandleProps {
  onPress?: () => void;
  width?: number;
  style?: ViewStyle;
}

export const AppHandle: React.FC<AppHandleProps> = ({
  onPress = () => null,
  width = 70,
  style,
}) => {
  // styles

  const styles = StyleSheet.create({
    container: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
    },
    handle: {
      width: width,
      height: 4,
      borderRadius: 3,
      backgroundColor: '#999',
      opacity: 0.35,
      marginRight: 'auto',
      marginLeft: 'auto',
    },
  });

  return (
    <View style={[styles.container, style]}>
      <TouchableOpacity
        style={{ width: '100%', padding: 10 }}
        onPress={onPress}
      >
        <View style={styles.handle} />
      </TouchableOpacity>
    </View>
  );
};
