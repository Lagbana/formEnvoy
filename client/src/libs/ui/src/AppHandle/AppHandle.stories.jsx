import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppHandle } from './AppHandle'
import { ViewStyle, StyleSheet } from 'react-native'

	const dummyStyles: ViewStyle = {}
  const styles = StyleSheet.create({
    container: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0
    },
    handle: {
      width: 70,
      height: 4,
      borderRadius: 3,
      backgroundColor: '#999',
      opacity: 0.35,
      marginRight: 'auto',
      marginLeft: 'auto'
    }
  })


storiesOf('AppHandle', module)
  .add('Default', () => (
    <AppHandle
			style={{...styles, ...dummyStyles}}
    />
  ))

