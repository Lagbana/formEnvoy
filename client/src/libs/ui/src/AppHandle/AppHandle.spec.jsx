import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppHandle } from './AppHandle'

configure({ adapter: new Adapter() })

test('<AppHandle /> renders correctly', () => {
  const component = shallow(<AppHandle />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
