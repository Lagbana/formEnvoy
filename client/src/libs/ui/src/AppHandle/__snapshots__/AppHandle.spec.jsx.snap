// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHandle /> renders correctly 1`] = `""`;
